cpp/common/src/codingstandards/cpp/types/Compatible.qll

```diff
@@ -331,26 +376,55 @@ module TypeEquivalence<TypeEquivalenceSig Config, TypeSubset T> {
   }
 
   bindingset[t1, t2]
-  private predicate equalTypedefTypes(RelevantTypedefType t1, RelevantTypedefType t2) {
+  private predicate equalTypedefTypes(TypedefType t1, TypedefType t2) {
     exists(Boolean baseTypesEqual |
       (baseTypesEqual = true implies equalTypes(t1.getBaseType(), t2.getBaseType())) and
       Config::equalTypedefTypes(t1, t2, baseTypesEqual)
     )
   }
 }
 
-module FunctionDeclarationTypeEquivalence<TypeEquivalenceSig Config> {
+signature predicate interestedInFunctionDeclarations(
+  FunctionDeclarationEntry f1, FunctionDeclarationEntry f2
+);
+
+module FunctionDeclarationTypeEquivalence<
+  TypeEquivalenceSig Config, interestedInFunctionDeclarations/2 interestedInFunctions>
+{
+  private predicate interestedInReturnTypes(Type a, Type b) {
+    exists(FunctionDeclarationEntry aFun, FunctionDeclarationEntry bFun |
+      interestedInFunctions(aFun, bFun) and
+      a = aFun.getType() and
+      b = bFun.getType()
+    )
+  }
+
+  private predicate interestedInParameterTypes(Type a, Type b) {
+    exists(FunctionDeclarationEntry aFun, FunctionDeclarationEntry bFun, int i |
+      interestedInFunctions(pragma[only_bind_into](aFun), pragma[only_bind_into](bFun)) and
+      a = aFun.getParameterDeclarationEntry(i).getType() and
+      b = bFun.getParameterDeclarationEntry(i).getType()
+    )
+  }
+
   predicate equalReturnTypes(FunctionDeclarationEntry f1, FunctionDeclarationEntry f2) {
```

Comment:
I think you need to add `interestedInFunctions(f1,f2)` here - the performance logs suggest this predicate is slow.